### ISSUE TITLE
vm crashes (exit code 139) with complex generics

### ISSUE THREAD

**Issue by Andersmholmgren** [dart-lang/sdk#25122]
If you run the below code in 1.14.0-dev.1.0 (note it's been crashing in several prior versions too) it crashes with

```
/usr/local/opt/dart/libexec/bin/dart --ignore-unrecognized-flags --checked --enable-vm-service:62159 --trace_service_pause_events /Users/blah/dart/backlogio_acdart/gitbacklog_root/gissue_root/gissue_client/lib/ui_components_2/issues/delme.dart
Observatory listening on http://127.0.0.1:62159

Process finished with exit code 139
```

I get no warnings from strong mode analyzer with this code however

```
main() {}

abstract class ComponentState<S extends ComponentState> {}

abstract class EntityState<ES extends EntityState> extends ComponentState<ES> {}

abstract class FrComponent<S extends ComponentState<S>> {}

abstract class AbstractListMember<E, M extends AbstractListMember<E, M>>
    extends ComponentState<M> {}

abstract class AbstractListEditorState<
    D,
    M extends AbstractListMember<D, M>,
    ES extends EntityState<ES>,
    S extends AbstractListEditorState<D, M, ES, S>> extends ComponentState<S> {}

abstract class AbstractListEditor<
    ES extends EntityState<ES>,
    E extends FrComponent<ES>,
    D,
    M extends AbstractListMember<D, M>,
    S extends AbstractListEditorState<D, M, ES, S>> extends FrComponent<S> {}

abstract class RepoListEditorState<M extends RepoListMember<M>,
        ES extends EntityState<ES>, S extends RepoListEditorState<M, ES, S>>
    extends AbstractListEditorState<String, M, ES, S>
    implements EntityState<S> {}

class RepoListMember<M extends RepoListMember<M>>
    extends AbstractListMember<String, M> {}

abstract class RepoListEditor<
        ES extends EntityState<ES>,
        E extends FrComponent<ES>,
        M extends RepoListMember<M>,
        S extends RepoListEditorState<M, ES, S>>
    extends AbstractListEditor<ES, E, String, M, S> {}
```

**Comment by fsc8000** [comment]
Thanks for the repro! I managed to reduce it to the following lines:
```dart
main() {}

abstract class ComponentState<S extends ComponentState> {}

abstract class AbstractListMember<E, M extends AbstractListMember<E, M>>
    extends ComponentState<M> {}

class RepoListMember<M extends RepoListMember<M>>
    extends AbstractListMember<String, M> {}
```

Results in the following crash:
```bash
$ out/DebugX64/dart crash.dart 
runtime/vm/object.cc:4522: error: expected: Length() >= (from_index + len)
Aborted (core dumped)
```

**Comment by fsc8000** [comment]
```dart
main() {}

class AbstractListMember<E, M extends AbstractListMember<E, M>> { }

class RepoListMember<M extends RepoListMember<M>>
    extends AbstractListMember<String, M> {}
```

**Comment by fsc8000** [comment]
@crelier

**Comment by Andersmholmgren** [comment]
thanks awesome Dart VM team. I can verify this is fixed in 1.14.0-dev.4.0

thanks for the quick turnaround. This will result in me pulling a lot less of my hair out
